sharder/deterministic.go

```diff
@@ -291,20 +294,101 @@ func (d *DeterministicSharder) loadPeerList() error {
 	} else {
 		d.peerLock.RUnlock()
 	}
+	// the random partitioning may have significant imbalances, so we try to
+	// balance them by moving partitions from the most heavily loaded nodes to
+	// the least heavily loaded nodes. But it's only possible if the number of
+	// partitions per peer is > 1.
+	if partitionsPerPeer > 1 {
```

Comment:
Does this also only need to happen if the number of refinery nodes is greater than 1?  If only a single instance of Refinery is running then balancing is unnecessary right?

Reply:
Good catch, yes.